apps/web/src/components/Badges/StorageBadge.tsx

```diff
@@ -21,31 +17,31 @@ type StorageConfig = {
 const STORAGE_CONFIGS: Record<BlobStorage, StorageConfig> = {
   file_system: {
     name: "File System",
-    icon: <ArchiveBoxIcon className="h-4 w-4" />,
+    icon: <StorageIcon storage="file_system" />,
     style:
       "bg-gray-100 hover:bg-gray-200 text-gray-800 hover:text-gray-900 dark:bg-gray-700 dark:text-gray-300 dark:hover:bg-gray-600 dark:hover:text-gray-200",
     buildDownloadUrl(_) {
       return "#";
     },
   },
   google: {
-    icon: <GoogleIcon />,
+    icon: <StorageIcon storage="google" />,
     style:
       "bg-slate-100 hover:bg-slate-200 text-slate-800 hover:text-slate-900 dark:bg-slate-700 dark:text-slate-300 dark:hover:bg-slate-600 dark:hover:text-slate-200",
     buildDownloadUrl(blobReference) {
-      return `https://storage.googleapis.com/${env.NEXT_PUBLIC_GOOGLE_STORAGE_BUCKET_NAME}/${blobReference}`;
+      return buildStorageDownloadUrl("google", blobReference) ?? "#";
     },
   },
   swarm: {
-    icon: <SwarmIcon />,
+    icon: <StorageIcon storage="swarm" />,
     style:
       "bg-orange-100 hover:bg-orange-200 text-orange-800 hover:text-orange-900 dark:bg-orange-900 dark:text-orange-300 dark:hover:bg-orange-800 dark:hover:text-orange-200",
     buildDownloadUrl(blobReference) {
-      return `https://gateway.ethswarm.org/access/${blobReference}`;
+      return buildStorageDownloadUrl("swarm", blobReference) ?? "#";
```

Comment:
According to buildStorageDownloadUrl() implementation, when first parameter is "swarm" it will always return a string so I guess we can safely remove the `?? "#"` part